projects/js-packages/publicize-components/src/components/media-section/index.js

```diff
@@ -70,63 +83,69 @@ export default function MediaSection() {
 	const { attachedMedia, updateAttachedMedia } = useAttachedMedia();
 	const { enabledConnections } = useSocialMediaConnections();
 
-	const { maxImageSize, getValidationError } = useMediaRestrictions( enabledConnections );
+	const { maxImageSize, getValidationError, getAllowedMediaTypes } = useMediaRestrictions(
+		enabledConnections
+	);
 	const allowedMediaTypes = getAllowedMediaTypes();
 
 	const mediaObject = useSelect(
-		select => select( 'core' ).getMedia( attachedMedia[ 0 ] || null, { context: 'view' } ),
+		select => select( 'core' ).getMedia( attachedMedia[ 0 ]?.id || null, { context: 'view' } ),
 		[ attachedMedia[ 0 ] ]
 	);
-	const { mediaWidth, mediaHeight, mediaSourceUrl, fullFileSize, mime } = getMediaDetails(
-		mediaObject
-	);
+
+	const { mediaData, metaData } = getMediaDetails( mediaObject );
 
 	useEffect( () => {
 		// Removes selected media if connection change results in invalid image
```

Comment:
Well, I think we should. Right now our validation stands on the principle, that we cannot create an invalid state in the editor. If the user plays around with the connections, that could create an invalid state, so we need to remove the media to avoid that.

We could change the approach and handle invalid states too, but in that case, we need to disable the publish button or something if that happens. I think I still like the current approach better. What do you think?

Reply:
More here: p1672959287715339-slack-C02JJ910CNL